### PR TITLE
Add Unincorporate to Domain/Clean Relation

### PR DESCRIPTION
**Pull request by srvasude**
Adds back methods for unincorporating in a domain or clean relation (which propagates to unincorporating correctly in irm and hirm).

Add some tests to check unincorporating works.

### CODE REVIEW

File: cxx/clean_relation_test.cc [hunk 1]
```diff
@@ -52,20 +52,56 @@ BOOST_AUTO_TEST_CASE(test_clean_relation) {
...
+  R2.set_cluster_assignment_gibbs(D2, 3, 1, &prng);
+  D1.set_cluster_assignment_gibbs(1, 1);
 
   Distribution<std::string>* db2 = R2.make_new_distribution(&prng);
   BOOST_TEST(db2->N == 0);
   db2->incorporate("hello");
   BOOST_TEST(db2->N == 1);
 }
+
+BOOST_AUTO_TEST_CASE(test_unincorporate) {
```
Comment (emilyfertig) [inline cxx/clean_relation_test.cc:80]: `NoisyRelation::unincorporate` calls `CleanRelation::unincorporate`, could you test NoisyRelation's or disable it? (NoisyRelation manages its own `data` map, so I think `NoisyRelation::unincorporate` needs to remove the item from `data`, which it doesn't currently do).

**Review by emilyfertig — CHANGES_REQUESTED** [review]
Looks good, thank you!

File: cxx/clean_relation_test.cc (hunk 1 above)
Comment (srvasude) [reply cxx/clean_relation_test.cc:80]: Done (and removed it from it's own data field).

**Review by emilyfertig — APPROVED** [review]
Thanks!

**Pull request merged**